cleanrl/sac_continuous_action_jax.py

```diff
@@ -0,0 +1,548 @@
+# docs and experiment results can be found at https://docs.cleanrl.dev/rl-algorithms/sac/#sac_continuous_actionpy
+# Implementation adapted from https://github.com/araffin/sbx
+import argparse
+import os
+import random
+import time
+from dataclasses import dataclass
+from distutils.util import strtobool
+from functools import partial
+from typing import Any, NamedTuple, Optional, Sequence
+
+import flax
+import flax.linen as nn
+import gym
+import jax
+import jax.numpy as jnp
+import numpy as np
+import optax
+import pybullet_envs  # noqa
+import tensorflow_probability
+from flax.training.train_state import TrainState
+from stable_baselines3.common.buffers import ReplayBuffer
+from stable_baselines3.common.env_util import make_vec_env
+from stable_baselines3.common.evaluation import evaluate_policy
+from stable_baselines3.common.vec_env import DummyVecEnv
+from torch.utils.tensorboard import SummaryWriter
+
+# Add progress bar if available
+try:
+    from tqdm.rich import tqdm
+except ImportError:
+    tqdm = None
+
+tfp = tensorflow_probability.substrates.jax
+tfd = tfp.distributions
+
+
+class RLTrainState(TrainState):
```

Comment:
Actually, what the motivation behind shadowing the `TrainState` class instead of using a different name?

for TD3, all networks have a frozen target, so it is okish even though not so clean, but for SAC, only the qf have target networks.

Reply:
Good point. I am changing the naming in other jax implementations as well.